lib/phoenix_live_view.ex

```diff
@@ -540,31 +545,49 @@ defmodule Phoenix.LiveView do
 
         # Ensures common `assigns` are applied to all LiveViews
         # that attach this module as an `on_mount` hook
-        def mount(_params, _session, socket) do
+        def on_mount(_params, _session, socket) do
           {:cont, assign(socket, :page_title, "DemoWeb")}
         end
+
+        # The following are all examples of passing additional
+        # arguments into a on_mount callback.
+
+        def on_mount_too(_, _, socket, :extra) do
+          {:cont, socket}
+        end
+
+        def on_mount_three(_, _, socket, role: :admin) do
+          {:cont, socket}
+        end
+
+        def on_mount_args(_, _, socket, :foo, :bar, :etc) do
+          {:cont, socket}
+        end
       end
 
       defmodule DemoWeb.PageLive do
         use Phoenix.LiveView
 
         on_mount {DemoWeb.LiveAuth, :ensure_mounted_current_user}
         on_mount DemoWeb.InitAssigns
+        on_mount {DemoWeb.InitAssigns, :on_mount_too, :extra}
+        on_mount {DemoWeb.InitAssigns, :on_mount_three, role: :admin}
```

Comment:
We should enforce the third argument to always be a list, otherwise there is an ambiguity if `[:foo, :bar, :etc]` are three separate args or a single argument. I understand we are trying to provide conveniences but I would stick with the regular mfargs rules without special cases for keywords, atoms, and so on.

Finally, I believe most APIs prepend the mfargs rather than append (it keeps the symmetry of . :)

Reply:
I am happy to remove the ambiguity– I assumed that would be the case, but I wanted to wait for feedback :)

And I would have sworn we append the args, but I'll make that change too!

Reply:
Done!